scripts/hid_configurator/modules/dfu.py

```diff
@@ -366,117 +415,142 @@ def file_crc(dfu_image):
     return crc32
 
 
-def dfu_sync_wait(dev, is_active):
-    if is_active:
-        dfu_state = 0x01
-    else:
-        dfu_state = 0x00
+def dfu_sync_wait_until_inactive(dev):
+    state_shown = False
 
-    for _ in range(DFU_SYNC_RETRIES):
+    while True:
         dfu_info = dfu_sync(dev)
+        if dfu_info is None:
+            break
 
-        if dfu_info is not None:
-            if dfu_info[0] != dfu_state:
-                # DFU may be transiting its state. This can happen when previous
-                # interrupted DFU operation is timing out. Sleep to allow it
-                # to settle the state.
-                time.sleep(DFU_SYNC_INTERVAL)
-            else:
-                break
+        if dfu_info.is_cleaning() and not state_shown:
+            print('Waiting for device to clean update slot')
+            state_shown = True
+
+        if dfu_info.is_busy():
```

Comment:
We may hang in never ending loop here

Reply:
That is true but should not be a problem as user can always stop the execution of the script. I was considering adding a timeout but it would have to be very long and the only place useful for it would be CI (which already should have its own timeout).